rfd/0202-db-multi-session-mfa.md

```diff
@@ -0,0 +1,309 @@
+    - Or, if first time requesting MFA for `multi-session` or shared `multi-session`
+      response is expired.
+    - No MFA if not required.
+  - Starts a local proxy in tunnel mode for this database (regardless of cluster
+    proxy listener mode).
+  - Craft a command for `os.exec`. Outputs are printed to `stdout` unless
+    `--log-dir` is specified.
+  - Execute the command.
+ 
+The command supports searching database by specifying one the following flags:
+- `--search`: List of comma separated search keywords or phrases enclosed in
+  quotations, e.g. `--search=foo,bar`.
+- `--labels`: List of comma separated labels to filter by labels, e.g.
+  `key1=value1,key2=value2`.
+- `--query` will NOT be supported for this command as it is harder to use than
+  the other options and the name `--query` may be confused with the database
+  query that needs to be executed.
+The command presents the search results then asks user to confirm before
+proceeding. `--skip-confirm` can be used to skip the confirmation.
+
+If no search parameters are used, a list of database names can be specified
+through `--dbs db1,db2,...`.
+
+For the MVP implementation, only PostgreSQL and MySQL databases will be
+supported. And a warning will be printed if the target databases have different
+protocols (e.g. `postgres` vs `mysql`).
+
+For databases that require per-session MFA, a prompt will still be presented
+per database.
+ 
+#### Possible future enhancements for `tsh db exec`
+- `tsh db exec --exec-config` to support a config file which allows specifying
+  different flags like `--db-user`, `--db-name`, `--exec-query` per target
+  database or per search.
+- `tsh db exec --command-template` to support custom command template like `tsh
+  db exec --exec-command "bash -c './my_script.sh {{.DB_SERVICE}} {{.DB_USER}}
+  {{.DB_NAME}} {{.DB_LOCAL_PORT}}'"`. An env var `TSH_UNSTABLE_DB_EXEC_COMMAND`
+  can be supported for the initial MVP.
+
+### Max connections
+
+The new `tsh db exec` command with the `--max-connections` flag makes it easier
+to for the client to flood the backend.
+
+A new role option is added to set a maximum number of in-flight connections:
+```diff
+kind: role
+version: v7
+spec:
+  options:
+    # limit number of concurrent Kubernetes sessions per user
+    max_kubernetes_connections: 1
++   # limit number of concurrent database sessions per user
++   max_db_connections: 5
+```
+
+The limit is enforced by Proxy using semaphore resources.
+
+`tsh db exec` will also ensure that value passed to `--max-connections` is
+smaller than the role option. If no value in the role set is specified, an
+arbitrary chosen value of 10 will be used as the max of the `--max-connections`
+flag.
+
+### Security
+
+There is no change regarding security for existing users, unless their Teleport
+admins set the `multi-session` mode in the role option.
+
+Since the mode is configured at the role level, the mode will only be applied to
```

Comment:
What about the global toggle under auth_service.authentication.require_session_mfa (or equivalent "cap" field)?